### ISSUE TITLE
Swin-Transformer 显存问题跟踪

### ISSUE THREAD

**Issue by Ldpe2G** [Oneflow-Inc/swin-transformer#6]
# Swin Transformer 8卡 DDP 显存占用对比

## 实验设置
实验机器 V100 16G
Pytorch 版本：    1.10.1
Oneflow 版本：   master nightly 0.6.0.dev20220102+cu112
Oneflow swin 测试分支： `swin_clean_ldp`
Pytorch swin 测试基于官方 swin 仓库

## 测试结果

| Framework | Swin Model | Batch Size per GPU | GPU0 显存占用 (单位MiB) |
| --------- | ---------- | ------------------ | ----------------------- |
| Pytorch   | Tiny 28M   | 64                 | 10264                   |
| Oneflow   | Tiny 28M   | 64                 | 11418                   |
| Pytorch   | Tiny 28M   | 96                 | 14054                   |
| Oneflow   | Tiny 28M   | 96                 | 15698                   |
| Pytorch   | Tiny 28M   | 128                | Out of Memory           |
| Oneflow   | Tiny 28M   | 128                | Out of Memory           |
| Pytorch   | Small 50M  | 64                 | 15054                   |
| Oneflow   | Small 50M  | 64                 | Out of Memory           |

  ## 测试结论

- Pytorch 上 batch 128 应该是用 32G 显存的显卡来训的，因为 swin tiny batch 128， 16G显卡也装不下

- 从最新的测试结果来看，oneflow 的显存占用单卡还是比 torch 高 1G 多

## 注
- Pytorch 在完整训练 swin base 的时候，会用上 accumulation steps 和 gradient checkpointing 等技术
